source/GameInterface/Services/Settlements/Patches/MobilePartyCachePatch.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Util;
+using GameInterface.Extentions;
+using GameInterface.Policies;
+using GameInterface.Services.Settlements.Messages;
+using HarmonyLib;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Settlements;
+
+namespace GameInterface.Services.Settlements.Patches;
+
+/// <summary>
+/// Handles Settlement AddMobileParty(MobileParty) and RemoveMobileParty(MobileParty) functions.
+/// </summary>
+[HarmonyPatch(typeof(Settlement))]
+public class MobilePartyCachePatch
+{
+
+    [HarmonyPatch("AddMobileParty")]
+    [HarmonyPrefix]
+    private static bool AddMobilePartyPrefix(ref Settlement __instance, ref MobileParty mobileParty)
+    {
+        if (AllowedThread.IsThisThreadAllowed()) return true;
+        if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+
+        if (ModInformation.IsClient) return false;
```

Comment:
Change this to match up to line 42 of https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/GameInterface/Services/Armies/Patches/ArmyLifetimePatches.cs#L42